src/tcgwars/logic/impl/gen4/LegendsAwakened.groovy

```diff
@@ -2412,8 +2420,8 @@ public enum LegendsAwakened implements LogicCardInfo {
             onAttack {
               damage 30
 
-              if (self.cards.filterByEnergyType(W)) {
-                def toDiscard = self.cards.filterByEnergyType(W).select(min: 0, max: 2, "For each [W] Energy discarded, do +10")
+              if (my.hand.filterByBasicEnergyType(W)) {
+                def toDiscard = my.hand.filterByBasicEnergyType(W).select(min: 0, max: 2, "For each [W] Energy discarded, do +10")
```

Comment:
It does not state "Basic Energy cards." Theoretically if a Special Energy card is released that provides [W] by default, this attack should be able to discard it.